core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -461,8 +510,9 @@ abstract class AbstractFetcherThread(name: String,
         val maybeTruncationComplete = fetchOffsets.get(topicPartition) match {
           case Some(offsetTruncationState) =>
             val state = if (offsetTruncationState.truncationCompleted) Fetching else Truncating
```

Comment:
Do we need to adjust this? I think we want to remain in the `Fetching` state if truncation detection is through `Fetch`.